frontend/appflowy_flutter/lib/window/app_window_size_manager.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:convert';
+
+import 'package:shared_preferences/shared_preferences.dart';
+
+class WindowSizeManager {
+  static const double minWindowHeight = 400.0;
+  static const double minWindowWidth = 600.0;
+
+  Future<void> saveSize(
+    double currentWindowHeight,
+    double currentWindowWidth,
+  ) async {
+    final windowSize = {
+      'height': currentWindowHeight < minWindowHeight
+          ? minWindowHeight
+          : currentWindowHeight,
+      'width': currentWindowWidth < minWindowWidth
+          ? minWindowWidth
+          : currentWindowWidth,
+    };
+
+    final prefs = await SharedPreferences.getInstance();
```

Comment:
I still think you should only do this once per instance. If you want some advice on how to achieve it using a future completer, let me know.

Reply:
Yes, I would appreciate your assistance with achieving that using a future completer. Thank you.